www/experiments.php

```diff
@@ -554,99 +554,170 @@ function refreshExperimentFormState(){
             echo "$id";
                              } ?>';
         if(priorState && currentTestID && priorState.indexOf("resubmit="+ currentTestID) > -1 ){
-            var form = $("form.experiments_grades");
-            form[0].reset();
+            var form = document.querySelector("form.experiments_grades");
+            form.reset();
             var pairs = priorState.split("&");
 
                 pairs.forEach(pair => {
                     var keyval = pair.split("=");
+                    if( keyval[0]){
+                        keyval[0] = decodeURIComponent(keyval[0]);
+                    }
+                    if( keyval[1]){
+                        keyval[1] = decodeURIComponent(keyval[1]);
+                    }
                     if( keyval[0] !== 'runs' ){
-                        var input = form.find("[name='" + keyval[0] + "']");
-
+                        var input = form.querySelectorAll("[type=checkbox][name='" + keyval[0] + "']");
                         if( input.length ){
-                            if( input.filter("[type=checkbox],[type=radio]").length ){
-                                input = input.filter( "[value='"+ keyval[1] +"']" ).attr("checked", true);
-                            } else if(input.filter("[type=text]").length ) {
-                                input.val(keyval[1]);
+                            input.forEach(inpt => {
+                                    if( inpt.value === keyval[1] ){
+                                        inpt.checked = true;
+                                    }
+                            });
+                        }
+                        var input = form.querySelector("textarea[name='" + keyval[0] + "']:not([data-hydrated='true']),input[type='text'][name='" + keyval[0] + "']:not([data-hydrated='true'])");
```

Comment:
I wonder if these queries will be more readable as templates

```js
form.querySelector(`textarea[name='${keyval[0]}']:not([data-...`);

Reply:
hmm yeah maybe? I left them for now... we could revisit